Automattic/wp-calypso
# ToS update: Add empty banner that retrieves legal updates data

Part of #39046

#### Changes proposed in this Pull Request

* Adds a `legal-updates-banner` feature flag that is currently enabled only in development.
* Adds a `LegalUpdatesBanner` component that will render in the future the ToS update notice. Currently it only retrieves data from the `wpcom/v2/legal` endpoint.
* Renders a `LegalUpdatesBanner` component in the Calypso layout if the `legal-updates-banner` feature flag is enabled.
* Adds data layer methods for retrieving data from the `wpcom/v2/legal` endpoint.

#### Next steps (follow-up PRs)
* Add selector that checks if user has accepted the current ToS.
* Display ToS update notice in `LegalUpdatesBanner` component if user has not accepted the current ToS.
* Add action for accepting current ToS.
* Dispatch action for accepting current ToS when user clicks on "Accept" in the `LegalUpdatesBanner` component.

#### Testing instructions

* Load any page in Calypso.
* Verify there is a request to `wpcom/v2/legal`.
<img width="901" alt="Screenshot 2020-01-28 at 16 46 28" src="https://user-images.githubusercontent.com/1233880/73279758-e730a180-41ed-11ea-91e9-723588f51e18.png">

* Verify the data is stored in `state.legal`:
<img width="224" alt="Screenshot 2020-01-28 at 16 51 41" src="https://user-images.githubusercontent.com/1233880/73280192-7b9b0400-41ee-11ea-9c9c-6c69b9777db5.png">
